src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -74,78 +73,165 @@ def handle_exception(ex):  # pylint: disable=too-many-return-statements, too-man
         error_msg = getattr(ex, 'message', str(ex))
         exit_code = 1
 
-        if isinstance(ex, AzCLIError):
+        if isinstance(ex, azclierror.AzCLIError):
```

Comment:
I still don't think `CLIError` should be categorized as `ValidationError`.

For example, if I `raise CLIError()` in [`show_version`](https://github.com/Azure/azure-cli/blob/38b14f949cbd85cd23f62fde355d3ca75584167f/src/azure-cli/azure/cli/command_modules/util/custom.py#L28-L31):

```py
def show_version(cmd):  # pylint: disable=unused-argument
    from knack.util import CLIError
    raise CLIError()
```

It is converted to a `ValidationError` which in fact is not:

```
> az version
ValidationError:
```

The same happens in https://github.com/Azure/azure-cli/issues/15320.

Reply:
I totally agree. 

CLIError is a very general error type, it can be any kind of error. Let's have some discussions in the error improvement meeting, to see how we deal with CLIError when applying the new error types. One option can be: 

NOT print error_type info for CLIError (also for other uncertain error types) until it is completely replaced.